### PR TITLE
feat(e2e): wait for pod to enter the Running state

### PR DESCRIPTION
**Pull request by bacongobbler**
Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>

### CODE REVIEW

File: tests/integration_tests.rs [hunk 1]
```diff
@@ -74,7 +76,40 @@ async fn test_wascc_provider() {
...
+        .expect("failed to poll for pod events")
+        .boxed();
+
+    while let Some(event) = watcher
+        .try_next()
+        .await
+        .expect("failed to poll for a pod event")
+    {
+        match event {
+            WatchEvent::Modified(o) => {
```
Comment (thomastaylor312) [inline tests/integration_tests.rs]: There should probably be a timeout here so that if it doesn't reach running that it doesn't keep going forever

Comment (bacongobbler) [reply tests/integration_tests.rs]: There's a ListParams timeout set above! Does that satisfy it, or is there another timeout we need to set?

**Comment by bacongobbler** [comment]
added more e2e tests to check the output of `kubectl logs`. Also verifies and closes #49

**Comment by bacongobbler** [comment]
Oh... I also refactored the tests to return a `Result`, cleaning up a lot of the code. Let me know if I should pull that out into a separate PR.

File: tests/integration_tests.rs (hunk 1 above)
Comment (thomastaylor312) [reply tests/integration_tests.rs]: Oh I skipped right over that part. I'm not used to list params having timeouts

File: crates/kubelet/src/server.rs [hunk 2]
```diff
@@ -122,11 +122,14 @@ async fn get_container_logs<T: Provider + Sync>(
     }
     match provider.logs(namespace, pod, container).await {
         Ok(data) => Response::new(Body::from(data)),
-        // TODO: This should detect not implemented vs. regular error
         Err(e) => {
             error!("Error fetching logs: {}", e);
-            let mut res = Response::new(Body::from("Not Implemented"));
-            *res.status_mut() = StatusCode::NOT_IMPLEMENTED;
+            let mut res = Response::new(Body::from(format!("Server error: {}", e)));
```
Comment (thomastaylor312) [inline crates/kubelet/src/server.rs]: Nit: This and the status should probably use the builder patter like we are doing elsewhere

Comment (bacongobbler) [reply crates/kubelet/src/server.rs]: Do you have a link to an example? Still wading my way around the codebase

**Pull request merged**